### ISSUE TITLE
Use a typed language

### ISSUE THREAD

**Issue by mchv** [guardian/alexa-skills-lambda#46]
Either:

- [ScalaJS](https://www.scala-js.org/)
- [Typescript 2.0](https://blogs.msdn.microsoft.com/typescript/2016/09/22/announcing-typescript-2-0/)

Using directly `scala` will be tempting but I think there is a latency cost for invoking lambda when the `jvm` has to start. It may not be immediate when testing with a single user but if usage growth the time to scale on `AWS` side may not be immediate

**Comment by LATaylor-guardian** [comment]
Yeah I was discussing this with Chris yesterday, think I might give ScalaJS a go.
